app/src/features/projects/view/components/Partnerships.tsx

```diff
@@ -57,29 +58,64 @@ const Partnerships: React.FC<IPartnershipsProps> = (props) => {
   } = props;
 
   const classes = useStyles();
-  const history = useHistory();
   const biohubApi = useBiohubApi();
 
   const [openEditDialog, setOpenEditDialog] = useState(false);
-  const [openErrorDialog, setOpenErrorDialog] = useState(false);
   const [partnershipsForUpdate, setPartnershipsForUpdate] = useState(ProjectPartnershipsFormInitialValues);
 
   const handleDialogEditOpen = async () => {
-    const { partnerships } = await biohubApi.project.getProjectForUpdate(id, [UPDATE_GET_ENTITIES.partnerships]);
+    let partnershipsResponseData;
 
-    if (!partnerships) {
-      setOpenErrorDialog(true);
+    try {
+      const response = await biohubApi.project.getProjectForUpdate(id, [UPDATE_GET_ENTITIES.partnerships]);
+
+      if (!response?.partnerships) {
+        showErrorDialog({ open: true });
+        return;
+      }
+
+      partnershipsResponseData = response.partnerships;
+    } catch (error) {
+      const apiError = new APIError(error);
+      showErrorDialog({ dialogText: apiError.message, open: true });
       return;
     }
 
-    setPartnershipsForUpdate(partnerships);
+    setPartnershipsForUpdate(partnershipsResponseData);
+
     setOpenEditDialog(true);
   };
 
-  const handleDialogEditSave = (values: IProjectPartnershipsForm) => {
-    // make put request from here using values and projectId
-    setOpenEditDialog(false);
-    history.push(`/projects/${id}/details`);
+  const handleDialogEditSave = async (values: IProjectPartnershipsForm) => {
+    const projectData = { partnerships: values };
+
+    try {
+      await biohubApi.project.updateProject(id, projectData);
+    } catch (error) {
+      const apiError = new APIError(error);
+      showErrorDialog({ dialogText: apiError.message, open: true });
```

Comment:
I was referring to a having a test for when we show the error dialog would be good.  If it's too hard, let's not worry about it for now.

Reply:
will look at it in subsequent PR